Failed local installation due to mismatched types compilation error

Hello,

I'm currently running into issues trying to install Martin locally using `cargo install`. 

```
error[E0308]: mismatched types
  --> /home/chrism/.cargo/registry/src/index.crates.io-6f17d22bba15001f/martin-0.13.0/src/fonts/mod.rs:83:39
   |
83 |         if face.get_char_index(cp) != 0 {
   |            -----------------------    ^ expected `Result<NonZero<u32>, Error>`, found integer
   |            |
   |            expected because this is `std::result::Result<std::num::NonZero<u32>, pbf_font_tools::freetype::Error>`
   |
   = note: expected enum `std::result::Result<std::num::NonZero<u32>, pbf_font_tools::freetype::Error>`
              found type `{integer}`
```

I've tested this on `rustc 1.78.0` stable and nightly, and on Martin v0.13, 0.12, and 0.11 on a fresh Ubuntu 22.04 install (via `cargo install --version=^0.11/12/13 martin`), all of which are thowing the same error. This appears to be the same sort of error mode reported at https://github.com/huggingface/text-generation-inference/issues/840 , although I don't believe Martin uses the `tokenizers` crate and I had trouble trying to find which dependency version might be the cause (maybe `pbf_font_tools`?). 

Thanks for your consideration, and I can provide any add'l information, just let me know!

--- a/.cargo-husky/hooks/pre-push
+++ b/.cargo-husky/hooks/pre-push
@@ -16,7 +16,7 @@ set -e
 if ! command -v just > /dev/null; then
     echo "Command 'just' is not installed. See https://github.com/casey/just#readme for installation instructions, or use"
     echo ""
-    echo "  cargo install just"
+    echo "  cargo install just --locked"
     echo ""
     exit 1
 fi

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -269,7 +269,7 @@ jobs:
         run: |
           set -x
           sudo apt-get install -y dpkg dpkg-dev liblzma-dev
-          cargo install cargo-deb
+          cargo install cargo-deb --locked
           cargo deb -v -p martin --output target/debian/debian-x86_64.deb
           mkdir -p target_releases
           mv target/debian/debian-x86_64.deb target_releases/

--- a/docs/src/development.md
+++ b/docs/src/development.md
@@ -30,7 +30,7 @@ sudo apt install -y  build-essential pkg-config jq file
 Install [Just](https://github.com/casey/just#readme) (improved makefile processor). Note that some Linux and Homebrew distros have outdated versions of Just, so you should install it from source:
 
 ```shell, ignore
-cargo install just
+cargo install just --locked
 ```
 
 When developing MBTiles SQL code, you may need to use `just prepare-sqlite` whenever SQL queries are modified. Run `just` to see all available commands.

--- a/docs/src/installation.md
+++ b/docs/src/installation.md
@@ -21,7 +21,7 @@ You can download martin from [GitHub releases page](https://github.com/maplibre/
 If you [install Rust](https://www.rust-lang.org/tools/install), you can build martin from source with Cargo:
 
 ```shell
-cargo install martin
+cargo install martin --locked
 martin --help
 ```
 

--- a/docs/src/tools.md
+++ b/docs/src/tools.md
@@ -12,6 +12,6 @@ Martin project contains additional tooling to help manage the data servable with
 
 Use `mbtiles --help` to see a list of available commands, and `mbtiles <command> --help` to see help for a specific command.
 
-This tool can be installed by compiling the latest released version with `cargo install mbtiles`, or by downloading a pre-built binary from the [releases page](https://github.com/maplibre/martin/releases/latest).
+This tool can be installed by compiling the latest released version with `cargo install mbtiles --locked`, or by downloading a pre-built binary from the [releases page](https://github.com/maplibre/martin/releases/latest).
 
 The `mbtiles` utility builds on top of the [MBTiles specification](https://github.com/mapbox/mbtiles-spec). It adds a few additional conventions to ensure that the content of the tile data is valid, and can be used for reliable diffing and patching of the tilesets.

--- a/justfile
+++ b/justfile
@@ -348,10 +348,10 @@ cargo-install $COMMAND $INSTALL_CMD="" *ARGS="":
     set -eu
     if ! command -v $COMMAND > /dev/null; then
         if ! command -v cargo-binstall > /dev/null; then
-            echo "$COMMAND could not be found. Installing it with    cargo install ${INSTALL_CMD:-$COMMAND} {{ ARGS }}"
-            cargo install ${INSTALL_CMD:-$COMMAND} {{ ARGS }}
+            echo "$COMMAND could not be found. Installing it with    cargo install ${INSTALL_CMD:-$COMMAND} --locked {{ ARGS }}"
+            cargo install ${INSTALL_CMD:-$COMMAND} --locked {{ ARGS }}
         else
-            echo "$COMMAND could not be found. Installing it with    cargo binstall ${INSTALL_CMD:-$COMMAND} {{ ARGS }}"
-            cargo binstall ${INSTALL_CMD:-$COMMAND} {{ ARGS }}
+            echo "$COMMAND could not be found. Installing it with    cargo binstall ${INSTALL_CMD:-$COMMAND} --locked {{ ARGS }}"
+            cargo binstall ${INSTALL_CMD:-$COMMAND} --locked {{ ARGS }}
         fi
     fi

